lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2584,7 +2584,7 @@ static ArrayRef<Decl *> evaluateMembersRequest(
   SmallVector<Decl *, 8> result;
 
   // If there's no parent source file, everything is already in order.
-  if (!dc->getParentSourceFile()) {
+  if (!dc->getParentSourceFile() || idc->getDecl()->isImplicit()) {
```

Comment:
This feels like it might break other kinds of synthesized conformances on implicit types, if we have those (or get them in the future).

Would it be better if the tangent vector struct synthesis didn't set isSynthesized() on the stored properties instead?

Reply:
I agree with Slava on this; this means that all implicit types would have their ABI defined by the order in which they got filled in, which is dangerous. (The bugs we triage from not sorting are dreadful miscompiles)

Reply:
> Would it be better if the tangent vector struct synthesis didn't set isSynthesized() on the stored properties instead?

Ok, thanks, I'll try that!

Reply:
That worked. Ready to re-review.